### PR TITLE
Dashboard: Bit of clean up for My Stories Karma tests

### LINKED ISSUE #8817: Dashboard: Separate 'My Stories' Karma tests, round out tests
We've got a lot of tests for the dashboard's 'my stories' page, which is awesome. It would be beneficial to separate out these into separate files and round out the coverage where applicable. I'm mostly thinking about testing focus of the grid/list view and more keyboard testing since we've done so much to make sure those are accessible. 

This is meant as a clean up task to make adding more tests easier in the future and to flesh out missing sections (retaining focus on context menus with keyboard, limited actions based on capabilities)

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

We've got a lot of tests for the my stories dashboard view, they all lived in the same file and updating things was getting hard to parse through. 

## Summary

Honestly, our coverage here is already good - I just dried up the code a little bit, added an aXe violation test for the list view (fixed 2 issues to make it pass), and updated some of the test names to be less duplicative. 

## User-facing changes

I darkened the link color on the list view of my stories because it wasn't passing contrast. Chose to darken the font color instead of lighten the background because we're already using the lightest shade of grey in the theme. 
Screen Shot 2021-10-22 at 5 00 49 PM

## Testing Instructions

If the dashboard karma tests still pass then this is good. 

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8817

### CODE REVIEW

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js [hunk 1]
```diff
@@ -111,152 +112,111 @@ describe('Grid view', () => {
...
-    let utils = within(firstStory);
+      utils = within(firstStory);
+      moreOptionsButton = utils.getByRole('button', {
+        name: /^Context menu for/,
+      });
 
-    const moreOptionsButton = utils.getByRole('button', {
-      name: /^Context menu for/,
+      await fixture.events.click(moreOptionsButton);
     });
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:151]: 💭 do you think it'd be helpful to add a small test verifying that clicking the `moreOptionsButton` opens the context menu?

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js [hunk 2]
```diff
@@ -393,12 +353,29 @@ describe('Grid view', () => {
...
+  });
 
-    it('should sort by Date Created', async () => {
-      const sortDropdown = fixture.screen.getByLabelText(
+  describe('Creator can sort their stories', () => {
+    let sortDropdown;
+    beforeEach(async () => {
+      sortDropdown = await fixture.screen.getByLabelText(
         'Choose sort option for display'
       );
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:363]: 🤔 Should this live in its own fixture file?

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js [hunk 3]
```diff
@@ -642,13 +590,14 @@ describe('Grid view', () => {
     });
 
     it('should exit the grid and re-focus the first item', async () => {
-      const storyCards = fixture.screen.getAllByTestId(/edit-story-grid-link/);
-      const firstStory = storyCards[0];
-      const lastStory = storyCards[storyCards.length - 1];
-      await focusOnGridByKeyboard();
+      const allItemGridLinks =
+        fixture.screen.getAllByTestId(/edit-story-grid-link/);
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:594]: Maybe this should live in its own fixture file too? Unless it's too much overhead. The fixture files seem to be the norm thought

File: packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js [hunk 4]
```diff
@@ -0,0 +1,613 @@
...
+      const listViewTable = fixture.screen.getByTestId('story-list-view');
+
+      expect(listViewTable).toBeTruthy();
+      const gridViewButton = fixture.screen.getByLabelText(
+        new RegExp(`^${VIEW_STYLE_LABELS[VIEW_STYLE.LIST]}$`)
+      );
+
+      await fixture.events.click(gridViewButton);
+
+      const storyElements = fixture.screen.getAllByTestId(/^story-grid-item/);
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js:102]: Nit: may be redundant, but to make this test a little sturdier there could be a check that tests to see that no story grid items are on the page before switching tabs

File: packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js [hunk 5]
```diff
@@ -0,0 +1,613 @@
...
+
+      const titleCell = utils.getByRole('heading', {
+        name: storiesSortedByModified[1].title,
+      });
+
+      await fixture.events.hover(titleCell);
+
+      const moreOptionsButton = utils.getByRole('button', {
+        name: /^Context menu for/,
+      });
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js:129]: I see this a lot in both of these files, and the right click menu karma tests. I don't remember exactly why I kept this constrained to that test file. Maybe we add a ticket to add all this stuff to its own fixture file?

File: packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+        // disable eslint to prevet overlapping .act calls
+        // eslint-disable-next-line no-await-in-loop
+        await fixture.events.keyboard.press('Backspace');
+      }
+
+      await fixture.events.keyboard.type('A New Title');
+
+      await fixture.events.keyboard.press('Enter');
+
+      expect(utils.getByText(/A New Title/)).toBeTruthy();
```
Comment (samwhale) [inline packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js]: Should this be more specific?

```suggestion
      expect(utils.getByText(/^A New Title$/)).toBeTruthy();
```

**Review by samwhale — APPROVED** [review]
This is so dope. Thanks for spending so much time on this tedious stuff

**Comment by samwhale** [comment]
Seeing one karma test fail but I don't think it's related
image

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js (hunk 2 above)
Comment (BrittanyIRL) [reply packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:363]: the dashboard fixture isn't set up the same as the editor, the editor's one view whereas here you have several different views so moving stuff to the fixture is gonna be different than in the editor. could probably set it up with queryBy but that feels like a bigger lift than just reorganizing and drying up 1 test file here.

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js (hunk 3 above)
Comment (BrittanyIRL) [reply packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:594]: samsies as above ^^

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js (hunk 2 above)
Comment (samwhale) [reply packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:363]: yeah agreed - okay cool

File: packages/dashboard/src/app/views/myStories/karma/myStories.karma.js (hunk 1 above)
Comment (BrittanyIRL) [reply packages/dashboard/src/app/views/myStories/karma/myStories.karma.js:151]: well, we're looking for `Context menu for` so i feel like that's fine for here. later we're checking that individual actions in the menu work as expected.

File: packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js (hunk 4 above)
Comment (BrittanyIRL) [reply packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js:102]: sure!

File: packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js (hunk 5 above)
Comment (BrittanyIRL) [reply packages/dashboard/src/app/views/myStories/karma/myStoriesList.karma.js:129]: yeah so `utils` is what's `within` a story in the fixture - so that everything in there is query-able, it's isolated in a `beforeEach` for the grid, while in the list view it's getting done at different points in the test so it was harder to dry up. could put it in a helper but that doesn't really make it any more or less readable so I left it like this.  

Definitely for stuff like this having a fixture we could just rely on would be handy, but it's just 1 file that that would better.
